app/models/alma_pod_records/alma_pod_job.rb

```diff
@@ -3,23 +3,25 @@ module AlmaPodRecords
   class AlmaPodJob < LibJob
     attr_reader :documents
 
-    def initialize(incoming_file_list: nil, file_pattern: '\.tar\.gz$', since: nil)
+    def initialize(incoming_file_list: nil, file_pattern: '\.tar\.gz$', since: nil, directory: nil)
       super(category: 'AlmaPodRecords')
       since ||= Rails.application.config.pod.days_to_fetch.days.ago
       @file_list = incoming_file_list || AlmaPodFileList.new(file_pattern: file_pattern, since: since)
+      @download_dir = Pathname.new(directory || Rails.application.config.pod.pod_record_path)
       clean_documents
     end
 
     def handle(data_set:)
-      write_files
+      write_files.each { |filename| AlmaPodSender.new(filename: filename).send }
```

Comment:
@sandbergja This seems a bit complex to have a yield to call a local method.   Is rubocop complaining that the method to complex? Why not just change the code to
```
    def send_files
      timestamp = Time.zone.now.strftime('%Y-%m-%d-%H-%M-%S')
      @documents.each_with_index do |document, index|
        filename = @download_dir + "pod_clean.#{timestamp}.#{index}.xml"
        File.write(filename, document.to_xml)
        AlmaPodSender.new(filename: filename).send
      end
    end
```

Reply:
Although if you are separating concerns or creating the file and sending them, then maybe two lines?